src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,553 @@
+
+/-- If a pair of measures `have_lebesgue_decomposition`, then `singular_part` chooses the
+measure from `have_lebesgue_decomposition`, otherwise it returns the zero measure. -/
+@[irreducible]
+def singular_part (μ ν : measure α) : measure α :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h).1 else 0
+
+/-- If a pair of measures `have_lebesgue_decomposition`, then `radon_nikodym_deriv` chooses the
+measurable function from `have_lebesgue_decomposition`, otherwise it returns the zero function. -/
+@[irreducible]
+def radon_nikodym_deriv (μ ν : measure α) : α → ℝ≥0∞ :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h).2 else 0
+
+lemma have_lebesgue_decomposition_spec {μ ν : measure α} (h : have_lebesgue_decomposition μ ν) :
+  measurable (radon_nikodym_deriv μ ν) ∧ (singular_part μ ν) ⊥ₘ ν ∧
+  μ = (singular_part μ ν) + ν.with_density (radon_nikodym_deriv μ ν) :=
+begin
+  rw [singular_part, radon_nikodym_deriv, dif_pos h, dif_pos h],
+  exact classical.some_spec h,
+end
+
+@[measurability]
+lemma measurable_radon_nikodym_deriv (μ ν : measure α) :
+  measurable $ radon_nikodym_deriv μ ν :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { exact (have_lebesgue_decomposition_spec h).1 },
+  { rw [radon_nikodym_deriv, dif_neg h],
+    exact measurable_zero }
+end
+
+lemma mutually_singular_singular_part (μ ν : measure α) :
+  singular_part μ ν ⊥ₘ ν :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { exact (have_lebesgue_decomposition_spec h).2.1 },
+  { rw [singular_part, dif_neg h],
+    exact mutually_singular.zero.symm }
+end
+
+lemma singular_part_le (μ ν : measure α) : singular_part μ ν ≤ μ :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { obtain ⟨-, -, h⟩ := have_lebesgue_decomposition_spec h,
+    conv_rhs { rw h },
+    exact measure.le_add_right (le_refl _) },
+  { rw [singular_part, dif_neg h],
+    exact measure.zero_le μ }
+end
+
+lemma with_density_radon_nikodym_deriv_le (μ ν : measure α) :
+  ν.with_density (radon_nikodym_deriv μ ν) ≤ μ :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { obtain ⟨-, -, h⟩ := have_lebesgue_decomposition_spec h,
+    conv_rhs { rw h },
+    exact measure.le_add_left (le_refl _) },
+  { rw [radon_nikodym_deriv, dif_neg h, with_density_zero],
+    exact measure.zero_le μ }
+end
+
+instance {μ ν : measure α} [finite_measure μ] :
+  finite_measure (singular_part μ ν) :=
+finite_measure_of_le μ $ singular_part_le μ ν
+
+instance {μ ν : measure α} [sigma_finite μ] :
+  sigma_finite (singular_part μ ν) :=
+sigma_finite_of_le μ $ singular_part_le μ ν
+
+instance {μ ν : measure α} [finite_measure μ] :
+  finite_measure (ν.with_density $ radon_nikodym_deriv μ ν) :=
+finite_measure_of_le μ $ with_density_radon_nikodym_deriv_le μ ν
+
+instance {μ ν : measure α} [sigma_finite μ] :
+  sigma_finite (ν.with_density $ radon_nikodym_deriv μ ν) :=
+sigma_finite_of_le μ $ with_density_radon_nikodym_deriv_le μ ν
+
+theorem eq_singular_part
```

Comment:
can you add a docstring to this one, explaining that this is part of the uniqueness of the Lebesgue decomposition (and give a reference for the other lemma giving the other part).